client/router_update.go

```diff
@@ -450,6 +456,35 @@ func (cli *VanClient) RouterUpdateVersionInNamespace(ctx context.Context, hup bo
 		kube.AppendSharedVolume(&router.Spec.Template.Spec.Volumes, []*[]corev1.VolumeMount{&router.Spec.Template.Spec.Containers[0].VolumeMounts, &router.Spec.Template.Spec.Containers[1].VolumeMounts}, "skupper-router-certs", "/etc/skupper-router-certs")
 		updateRouter = true
 	}
+	if addClaimsSupport {
+		err = kube.UpdateRole(namespace, types.TransportRoleName, types.TransportPolicyRule, cli.KubeClient)
+		if err != nil {
+			return false, err
+		}
+		if !config.IsEdge() {
+			if usingRoutes {
+				err = cli.createClaimsRedemptionRoute(ctx, namespace)
+				if err != nil {
+					return false, err
+				}
+			}
+		}
+	}
+	if moveClaims {
+		if err := cli.moveClaimsToRouterService(ctx, namespace); err != nil {
+			return false, err
+		}
+		latest, err := cli.KubeClient.AppsV1().Deployments(namespace).Get(context.TODO(), types.TransportDeploymentName, metav1.GetOptions{})
```

Comment:
One thing I have noticed, while testing the update process, is that after the `regenerateSiteSecret`
completes (called by `moveClaimsToRouterService`), and the router is restarted, the
`skupper-router` deployment gets updated a few times, due to the statuses transitions, so the `latest`
resource obtained here, is not actually the latest, so I faced errors resulting from updates to a modified
skupper-router deployment quite a few times.

I tried adding a `WaitDeploymentReady` after `regenerateSiteSecret` is called and it seems to resolve it.

Reply:
Good suggestion, I have made that change.